src/firmware/app/control/python_controller/generate_difference_equation.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+
+import sys
+import control as ct
+from control.matlab import *
+
+CONTROLLER_DIFFERENCE_EQUATION_COEFFICIENTS = """
+#define WHEEL_SPEED_CONTROLLER_TICK_TIME 1e7
+#define GENERATED_NUM_WHEEL_SPEED_COEFFICIENTS {numerator_length};
+#define GENERATED_NUM_PREVIOUS_WHEEL_SPEED_COMMAND_COEFFICIENTS {denominator_length};
+float GENERATED_WHEEL_SPEED_COEFFICIENTS[GENERATED_NUM_WHEEL_SPEED_COEFFICIENTS] =
+{{{numerator_coefficients}}};
```

Comment:
Why do some variables have single `{}` and some have triple `{}`?

Reply:
`numerator_length` and `denominator_length` need only one bracket because we're just replacing them with the `format` method.
`numerator_coefficients` and `denominator_coefficients` need 3 brackets. This is because we need one pair of brackets for the `format` method. Another pair of brackets is because we need to initialize the array (`int arr[3] = {1,2,3}` for instance). The last pair is like an 'escape' for the pair of brackets. 
tl;dr one pair to do the format, one pair to initialize array, one pair to escape

Reply:
can we add a comment to that effect?

Reply:
Done.